internal/appsec/remoteconfig.go

```diff
@@ -228,17 +293,25 @@ func (a *appsec) enableRemoteActivation() error {
 	}
 	a.registerRCProduct(rc.ProductASMFeatures)
 	a.registerRCCapability(remoteconfig.ASMActivation)
-	a.registerRCCallback(a.asmFeaturesCallback, rc.ProductASMFeatures)
 	return nil
 }
 
-func (a *appsec) enableRCBlocking(handle wafHandleWrapper) error {
+func (a *appsec) enableRCBlocking() error {
 	if a.rc == nil {
 		return fmt.Errorf("no valid remote configuration client")
 	}
+
+	a.registerRCProduct(rc.ProductASM)
+	a.registerRCProduct(rc.ProductASMDD)
 	a.registerRCProduct(rc.ProductASMData)
-	a.registerRCCapability(remoteconfig.ASMIPBlocking)
 	a.registerRCCapability(remoteconfig.ASMUserBlocking)
-	a.registerRCCallback(handle.asmDataCallback, rc.ProductASMData)
+	a.registerRCCapability(remoteconfig.ASMRequestBlocking)
+
+	if _, isSet := os.LookupEnv(rulesEnvVar); !isSet {
+		a.registerRCCapability(remoteconfig.ASMIPBlocking)
+		a.registerRCCapability(remoteconfig.ASMDDRules)
+		a.registerRCCapability(remoteconfig.ASMExclusions)
```

Comment:
AFAIK, I think this list is incomplete and that more capabilities are conditioned by `DD_APPSEC_RULES`, such as user blocking and request blocking.

Also, shouldn't we conditionally register the RC products too to avoid advertising them when the capabilities are not set?

Reply:
Not according to https://datadoghq.atlassian.net/wiki/spaces/APS/pages/2844267422/ASM+DD+-+Rules+Updates. This may come to change later on but right now at least dotnet does it this way too.

As for the products, I guess we can do this for ASM_DD, for the rest, we can still need those configs while not having some specific capabilities (rules data _could_ be used for something other than blocking, ASM configs could relay custom rules and so on)

Reply:
Fixed in cc5abf5b67e133a